tpu-provisioner/internal/controller/creation_controller.go

```diff
@@ -96,5 +88,17 @@ func (r *CreationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 func (r *CreationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Pod{}).
+		WithEventFilter(predicate.NewPredicateFuncs(func(object client.Object) bool {
+			// Only reconcile pods which meet the conditions defined below.
+			pod, ok := object.(*corev1.Pod)
+			return ok &&
```

Comment:
Sure, added a check for this and a new test case for it as well.

Also introduced a constants.go file to the controller and moved kubebuilder marker for events RBAC to the creation controller file instead of the empty events.go file (renamed to constants.go).